CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+# 1. Requirements
+project(usdOpenAssetIOResolver)
+cmake_minimum_required(VERSION 3.21)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+#-----------------------------------------------------------------------
+# Default install directory
+
+# Default install to a `dist` directory under the build directory, ready
+# for use in tests and for packaging. But don't override if user has
+# explicitly set CMAKE_INSTALL_PREFIX.
+if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT AND PROJECT_IS_TOP_LEVEL)
+    set(CMAKE_INSTALL_PREFIX "${PROJECT_BINARY_DIR}/dist" CACHE PATH "Installation location" FORCE)
+endif ()
+
+add_compile_options(-Wno-deprecated)  # This is just to avoid some spammy warnings
```

Comment:
Yes, the specific error is.
```
In file included from /usr/include/c++/9/ext/hash_set:60,
                 from /usr/local/USD/include/pxr/base/tf/hashset.h:39,
                 from /usr/local/USD/include/pxr/base/tf/token.h:37,
                 from /usr/local/USD/include/pxr/base/vt/types.h:36,
                 from /usr/local/USD/include/pxr/base/vt/value.h:48,
                 from /usr/local/USD/include/pxr/usd/ar/assetInfo.h:31,
                 from /home/elliotmorris/data/usdOpenAssetIOResolver/elliotcmorris/src/resolver.cpp:8:
/usr/include/c++/9/backward/backward_warning.h:32:2: warning: #warning This file includes at least one deprecated or antiquated header which may be removed without further notice at a future date. Please use a non-deprecated interface with equivalent functionality instead. For a listing of replacement headers and interfaces, consult the file backward_warning.h. To disable this warning use -Wno-deprecated. [-Wcpp]
```

Reply:
Best practice would be to use `target_compile_options(${PLUGIN_NAME} PRIVATE -Wno-deprecated)` in the CMakeLists under `src`, so that this flag isn't applied globally to every single target even if it doesn't need it.

Reply:
Ahh shame, thanks!